opensearch-project/OpenSearch-Dashboards
# [Multiple Datasource] Allow top nav menu to mount data source menu for use case when both menus are mounted

### Description
This change allows top nav menu to mount data source menu for use case when both menus are mounted.

### Issues Resolved
Fixes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6229

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/17714150/03d0b154-77d4-46ba-922d-e8a5dddd2b56



## Testing the changes
The following steps were performed in the recording above:
1. enable data source plugin
2. go to maps and click on a map page and will see save button and data source menu mounted to navigation bar 

### Check List
